xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -592,11 +592,21 @@ bool CWinSystemWayland::SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, boo
       ApplyBufferScale(m_scale);
     }
 
-    // FIXME This assumes that the resolution has already been set. Should
-    // be moved to some post-change callback when resolution setting is refactored.
-    if (!m_inhibitSkinReload)
+    if (m_shellSurfaceState.test(IShellSurface::STATE_RESIZING) && !m_resizeSkinReloadTimer.IsRunning())
     {
-      g_application.ReloadSkin();
+      // Reloading the skin is costly -> during resizing (when the size changes quite
```

Comment:
I just tried resizing on OSX. ReloadSkin is never triggered. Maybe there is an issue with aspect ratio on resize but this can be solved without reloading skin.